posthog/models/organization.py

```diff
@@ -312,6 +312,48 @@ def validate_update(
             if membership_being_updated.level > self.level:
                 raise exceptions.PermissionDenied("You can only edit others with level lower or equal to you.")
 
+    def get_scoped_api_keys(self):
+        """
+        Get API keys that are scoped to this organization or its teams.
+        Returns a dictionary with information about the keys.
+        """
+        from posthog.models.personal_api_key import PersonalAPIKey
+        from django.db.models import Q
+        from posthog.models.team.team import Team
+        from django.utils import timezone
+
+        # Get teams that belong to this organization
+        team_ids = list(Team.objects.filter(organization_id=self.organization_id).values_list("id", flat=True))
+
+        # Find API keys scoped to either the organization or any of its teams
+        # Also include keys with no scoped teams or orgs (they apply to all orgs/teams)
+        personal_api_keys = PersonalAPIKey.objects.filter(user=self.user).filter(
+            Q(scoped_organizations__contains=[str(self.organization_id)])
+            | Q(scoped_teams__overlap=team_ids)
+            | (Q(scoped_organizations__isnull=True) | Q(scoped_organizations=[]) | Q(scoped_organizations=[]))
+            & (Q(scoped_teams__isnull=True) | Q(scoped_teams=[]) | Q(scoped_teams=[]))
+        )
+
+        # Get keys with more details
+        keys_data = []
+        has_keys = personal_api_keys.exists()
+
+        # Check if any keys were used in the last week
+        one_week_ago = timezone.now() - timedelta(days=7)
+        keys_active_last_week = personal_api_keys.filter(last_used_at__gte=one_week_ago).exists()
+
+        # Get detailed information about each key
+        for key in personal_api_keys:
+            keys_data.append({"name": key.label, "last_used_at": key.last_used_at})
+
+        return {
+            "personal_api_keys": personal_api_keys,
+            "has_keys": has_keys,
+            "keys_active_last_week": keys_active_last_week,
```

Comment:
Nit: this name sounds like as list rather than a bool, for a bool it should be something like `has_keys_active_last_week`

Reply:
Good call, can update.